twilight-model/src/application/interaction/metadata.rs

```diff
@@ -0,0 +1,25 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    id::{
+        marker::{GuildMarker, InteractionMarker, MessageMarker, UserMarker},
+        AnonymizableId, Id,
+    },
+    oauth::ApplicationIntegrationMap,
+};
+
+use super::InteractionType;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct InteractionMetadata {
+    pub id: Id<InteractionMarker>,
+    #[serde(rename = "type")]
+    pub kind: InteractionType,
+    pub user_id: Id<UserMarker>,
+    pub authorizing_integration_owners:
+        ApplicationIntegrationMap<AnonymizableId<GuildMarker>, Id<UserMarker>>,
```

Comment:
do we really need the complexity of generics here, we could follow ddocs and create a `AuthorizingIntegrationOwners` struct with fields `kind: ApplicationIntegrationType` and `id: AuthorizingIntegrationOwnerId` where `AuthorizingIntegrationOwners` is an enum with `Guild(Option<Id<GuildMarker>>)` and `User(Id<UserMarker>)`

Reply:
It is not a enum, instead if we should do it as in the docs we should have a hashmap with 2 possible enum keys and then different Id type depending on if it is one or the other, I think this is the easiest way to make it work nicely.

Reply:
i think we're adding complexity to be DRY here, i think it's worth a few added LOC to separate `ApplicationIntegrationMap` to different structs such as:
```rs
// for InteractionMetadata and Interaction
enum AuthorizingIntegrationOwners {
    GuildInstall(AuthorizingIntegrationOwnersGuild),
    UserInstall(Id<UserMarker>),
}

enum AuthorizingIntegrationOwnersGuild {
    Guild(Id<GuildMarker>),
    Dm,
}
```

and for `ApplicationIntegrationTypeConfig`, unless im missing anything we only ever use it as  `ApplicationIntegrationMap<ApplicationIntegrationTypeConfig>` so we can just unnest it